### PR TITLE
Use G4OpticalParameters to set Cerenkov/Scintillation parameters

### PR DESCRIPTION
**Pull request by wdconinc**
This introduces the use of G4OpticalParameters for setting of optical processes, cerenkov, and scintillation since the former approach ceased working as of geant4.10.7.

BEGINRELEASENOTES
- Use G4OpticalParameters in geant4.10.7 and newer

ENDRELEASENOTES

### CODE REVIEW

**Comment by wdconinc** [comment]
Addresses #868

File: DDG4/plugins/Geant4OpticalPhotonPhysics.cpp [hunk 1]
```diff
@@ -81,11 +86,18 @@ namespace dd4hep {
         G4OpRayleigh*         fRayleighScatteringProcess = new G4OpRayleigh();
         G4OpMieHG*            fMieHGScatteringProcess    = new G4OpMieHG();
 
+#if G4VERSION_NUMBER >= 1070
+        G4OpticalParameters* params = G4OpticalParameters::Instance();
+        params->SetAbsorptionVerboseLevel(m_verbosity);
+        params->SetRayleighVerboseLevel(m_verbosity);
+        params->SetMieVerboseLevel(m_verbosity);
+        params->SetBoundaryVerboseLevel(m_verbosity);
```
Comment (andresailer) [inline DDG4/plugins/Geant4OpticalPhotonPhysics.cpp:94]: Could you add the boundaryInvokeSD parameter setting to this?

https://github.com/Geant4/geant4/blob/6399a014b611a4aa09ecba225418222dac261ca1/source/processes/electromagnetic/utils/src/G4OpticalParameters.cc#L129

File: DDG4/plugins/Geant4ScintillationPhysics.cpp [hunk 2]
```diff
@@ -80,6 +84,15 @@ namespace dd4hep {
...
              yes_no(m_trackSecondariesFirst));
         G4Scintillation* process = new G4Scintillation(name());
+#if G4VERSION_NUMBER >= 1070
+        G4OpticalParameters* params = G4OpticalParameters::Instance();
+        params->SetScintVerboseLevel(m_verbosity);
+        params->SetScintFiniteRiseTime(m_finiteRiseTime);
+        params->SetScintStackPhotons(m_stackPhotons);
+        params->SetScintTrackSecondariesFirst(m_trackSecondariesFirst);
+        params->SetScintYieldFactor(m_scintillationYieldFactor);
+        params->SetScintExcitationRatio(m_scintillationExcitationRatio);
```
Comment (andresailer) [inline DDG4/plugins/Geant4ScintillationPhysics.cpp:94]: Could you add scintByParticleType and scintTrackInfo parameters?
https://github.com/Geant4/geant4/blob/6399a014b611a4aa09ecba225418222dac261ca1/source/processes/electromagnetic/utils/src/G4OpticalParameters.cc#L110-L111

**Review by andresailer — COMMENTED** [review]
Thank you for these fixes!

File: DDG4/plugins/Geant4ScintillationPhysics.cpp [hunk 3]
```diff
@@ -80,14 +86,32 @@ namespace dd4hep {
...
+        params->SetScintByParticleType(m_byParticleType);
+        params->SetScintTrackInfo(m_trackInfo);
+#else
         process->SetVerboseLevel(m_verbosity);
         process->SetFiniteRiseTime(m_finiteRiseTime);
-#if G4VERSION_NUMBER>1030
+#if G4VERSION_NUMBER >= 1030
```
Comment (wdconinc) [inline DDG4/plugins/Geant4ScintillationPhysics.cpp:102]: StackPhotons is available in 10.3.0.

File: DDG4/plugins/Geant4OpticalPhotonPhysics.cpp (hunk 1 above)
Comment (wdconinc) [reply DDG4/plugins/Geant4OpticalPhotonPhysics.cpp:95]: Implemented in 8e07cb7f9aa3b1c67aa607ab6bd6dbf6a0bcdd7a.

File: DDG4/plugins/Geant4ScintillationPhysics.cpp (hunk 2 above)
Comment (wdconinc) [reply DDG4/plugins/Geant4ScintillationPhysics.cpp:96]: Implemented in 8e07cb7f9aa3b1c67aa607ab6bd6dbf6a0bcdd7a.

**Review by andresailer — APPROVED** [review]
Thanks for implementing the additional options as well!

**Pull request merged**